### PR TITLE
[SPARK-41659][CONNECT] Enable doctests in pyspark.sql.connect.readwriter

### PR DESCRIPTION
**Pull request by techaddict**
### What changes were proposed in this pull request?
This PR proposes to enable doctests in pyspark.sql.connect.readwriter that is virtually the same as pyspark.sql.readwriter.

### Why are the changes needed?
To make sure on the PySpark compatibility and test coverage.

### Does this PR introduce any user-facing change?
No, doctest's only.

### How was this patch tested?
New Doctests Added

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
@zhengruifeng i think we should cover dictionary cases too (like we did in https://github.com/apache/spark/pull/39313):

```
Failed example:
    with tempfile.TemporaryDirectory() as d:
        # Write a DataFrame into a JSON file
        spark.createDataFrame(
            [{"age": 100, "name": "Hyukjin Kwon"}]
        ).write.mode("overwrite").format("json").save(d)

        # Read the JSON file as a DataFrame.
        spark.read.format('json').load(d).show()
Expected:
    +---+------------+
    |age|        name|
    +---+------------+
    |100|Hyukjin Kwon|
    +---+------------+
Got:
    +---+------------+
    | _1|          _2|
    +---+------------+
    |100|Hyukjin Kwon|
    +---+------------+
```

A lot of test fails because of that

**Comment by HyukjinKwon** [comment]
Filed a JIRA: SPARK-41746

**Comment by HyukjinKwon** [comment]
After https://github.com/apache/spark/pull/39344 is merged, most of them should pass.

File: python/pyspark/sql/connect/readwriter.py [hunk 1]
```diff
@@ -458,3 +459,61 @@ def orc(
...
+        import pyspark.sql.connect.readwriter
+
+        globs = pyspark.sql.connect.readwriter.__dict__.copy()
+        # Works around to create a regular Spark session
+        sc = SparkContext("local[4]", "sql.connect.readwriter tests", conf=SparkConf())
+        globs["_spark"] = PySparkSession(
+            sc, options={"spark.app.name": "sql.connect.readwriter tests"}
+        )
+
+        # SPARK-41746: infer schema for dictionaries
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/readwriter.py:483]: I merged this. Mind removing them and rebasing it to see if it passes?

Comment (techaddict) [reply python/pyspark/sql/connect/readwriter.py]: done, most passed 👍🏼

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**